contracts/src/L1/rollup/IT1Chain.sol

```diff
@@ -118,8 +122,16 @@ interface IT1Chain {
     /// @notice t1 batch finalization
     ///
     /// @param withdrawRoot The withdraw trie root of current batch.
-    /// @param signature The ECDSA valid signature to match one of the provers
-    function finalizeBatchWithProof(bytes32 withdrawRoot, bytes calldata signature) external;
+    /// @param withdrawRootSignature The ECDSA valid signature of withdraw root to match one of the provers
+    /// @param proofOfFill7683Root 7683 The Proof of Fill trie root of current batch.
+    /// @param proofOfFillRootSignature The ECDSA valid signature of proof of fill root to match one of the provers
+    function finalizeBatchWithProof(
```

Comment:
WDYT about overloading `finalizeBatchWithProof()` instead of modifying it? I haven't checked the contract size.

Reply:
I agree with Diego's suggestion, but let's not block this further.